File: weather-mvp-dagger/src/main/java/com/baronzhang/android/weather/di/ApplicationComponent.java
package com.baronzhang.android.weather.di;

import android.content.Context;

import com.baronzhang.android.weather.main.WeatherApplication;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author 张磊 (baron[dot]zhanglei[at]gmail[dot]com)
 *         2016/11/30
 */
@Singleton
@Component(modules = {ApplicationModule.class})
public interface ApplicationComponent {

    WeatherApplication getApplication();

    Context getContext();
}

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/di/ApplicationComponent.kt
package com.minimalist.weather.kotlin.di


import android.content.Context
import com.minimalist.weather.kotlin.main.WeatherApplication
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {

    val application: WeatherApplication

    val context: Context
}
